icaruscode/PMT/OpReco/ICARUSFlashAssAna_module.cc

```diff
@@ -0,0 +1,457 @@
+    m_abs_start_time = ophit->PeakTimeAbs();
+    m_pe = ophit->PE();
+    m_fast_to_total = ophit->FastToTotal();
+
+    ophittree->Fill();
+
+  }
+
+  m_multiplicity_left = std::accumulate( sumpe_map.begin(), sumpe_map.end(), 0,
+					 [&](int value, const std::map<int, float>::value_type& p) {
+					   return getSideByChannel(p.first)==0 ? ++value : value ;
+					 });
+
+  m_multiplicity_right =std::accumulate( sumpe_map.begin(), sumpe_map.end(), 0,
+					 [&](int value, const std::map<int, float>::value_type& p) {
+					   return getSideByChannel(p.first)==1 ? ++value : value ;
+					 });
+
+  m_sum_pe_left = std::accumulate( sumpe_map.begin(), sumpe_map.end(), 0.0, 
+				   [&](float value, const std::map<int, float>::value_type& p) {
+				     return getSideByChannel(p.first)==0 ? value+p.second : value ; 
+				   });
+  
+  m_sum_pe_right = std::accumulate( sumpe_map.begin(), sumpe_map.end(), 0.0,
+            [&](float value, const std::map<int, float>::value_type& p) {
+              return getSideByChannel(p.first)==1 ? value+p.second : value ;
+            }); 
+
+  //for( int i=0; i<3; i++ ){ xyz[i] /= (m_sum_pe_left+ m_sum_pe_right); }
+  
+}
+
+
+void opana::ICARUSFlashAssAna::endJob() {
+
+}
+
+
+void opana::ICARUSFlashAssAna::analyze(art::Event const& e) {
+
+
+  m_run = e.id().run();
+  m_event = e.id().event();
+  m_timestamp = e.time().timeHigh(); // precision to the second 
+
+
+  // We work out the trigger information here 
+  if( !fTriggerLabel.empty() ) { 
+
+      art::Handle<std::vector<sim::BeamGateInfo>> beamgate_handle;
+      e.getByLabel( fTriggerLabel, beamgate_handle );
+
+      if( beamgate_handle.isValid() ) {
+
+        for( auto const & beamgate : *beamgate_handle ) {
+
+          m_beam_gate_start = beamgate.Start(); 
+          m_beam_gate_width = beamgate.Width(); 
+          m_beam_type = beamgate.BeamType() ;
+
+        }
+
+      }
+
+      else {
+        std::cout << "Invalid Trigger Data product " << fTriggerLabel.label() << "\n" ;
+      }
+
+  }
+
+  else {
+     std::cout << "Trigger Data product " << fTriggerLabel.label() << " not found!\n" ;
+  }
+
+
+  // Now we take care of the labels
+  if ( !fFlashLabels.empty() ) {
+
+    for ( size_t i=0; i<fFlashLabels.size(); i++  ) {
+
+      auto const label = fFlashLabels[i];
+
+      art::Handle<std::vector<recob::OpFlash>> flash_handle;
+      e.getByLabel( label, flash_handle );
+
+      if( flash_handle.isValid() ) {
+
+        art::FindManyP<recob::OpHit> ophitsPtr( flash_handle, e, label );
+
+        for ( size_t idx=0; idx<flash_handle->size(); idx++ ) {
+
+          m_flash_id = idx;
+          auto const & flash = (*flash_handle)[idx];
+
+          m_flash_time = flash.Time();
+          m_sum_pe = flash.TotalPE();
+            
+          auto const & ophits = ophitsPtr.at(idx);
+
+          // Get the multiplicity, the position and the number of PE per Side
+          float xyz[3] = {0.0, 0.0, 0.0};
+          processOpHits( ophits, 
```

Comment:
What about OpHits that aren't in a flash? Should those not be saved?

Also, it looks like the hit information does not include the data you would need to match hits to flashes in the TTree? I believe that would also be useful.